PGo AutoTrader.ahk

```diff
@@ -0,0 +1,238 @@
+	; "--clipboard" flag is the way we tell it to do OCR and put the results on the clipboard
+	RunWait, %CONSTANT_CAPTURE_2_TEXT_EXECUTABLE_ABSOLUTE_PATH% -s "%theTopLeftX% %theTopLeftY% %theBottomRightX% %theBottomRightY%" --clipboard
+	clipboard := trim(clipboard,"`n`r`t ")
+	myReturn := clipboard
+	return myReturn
+}
+
+EXECUTE_SEQUENCE_STEP(theStepList, the1IndexedStepOrdinal, theIsOriginalTryFlag){
+	myExecuteResult := false
+	myStep := theStepList[the1IndexedStepOrdinal]
+
+	; use a local copy of the checkExistsIntervalList, since we may make changes
+	myCheckExistsIntervalList := myStep.getCheckExistsTryLimitIntervalList().clone()
+
+	if(!theIsOriginalTryFlag) {
+		myCheckExistsIntervalList := []
+		myCheckExistsIntervalList.Push(1)
+		;DISPLAY_MESSAGE("Overwrote checkExistsTryLimitIntervalList")
+	}
+
+	myCheckTextResult := false
+	myCheckExistsIntervalCount := myCheckExistsIntervalList.Length()
+	Loop, %myCheckExistsIntervalCount% {
+		; Reminder: "A_Index" is the built-in AutoHotkey index for loops
+		; https://www.autohotkey.com/docs/v1/Variables.htm#Index
+		myCheckExistsTryLimit := myCheckExistsIntervalList[A_Index]
+		myCheckTextResult := GENERALIZED_CHECK_TEXT_ON_SCREEN(myStep.getTextCheck().getSearchText(), myStep.getMillisecondsBetweenRetries(), myCheckExistsTryLimit, myStep.getTextCheck())
+		if(myCheckTextResult){
+			break
+		} else {
+			myIsLastIntervalFlag := A_Index == myCheckExistsIntervalCount
+			if(theIsOriginalTryFlag AND !myIsLastIntervalFlag){
+				;DISPLAY_MESSAGE("Retrying previous step")
+				myPreviousStepOrdinal := GET_PREVIOUS_STEP_ORDINAL(the1IndexedStepOrdinal, theStepList.Length())
+				myRetryPreviousResult := EXECUTE_SEQUENCE_STEP(theStepList, myPreviousStepOrdinal, false)
+				;DISPLAY_MESSAGE("Retried previous step, got " . myRetryPreviousResult)
+				HANDLE_GLOBAL_INTERRUPT_ALL()
+			}
+		}
+	}
+
+	if(myCheckTextResult){
+		myExecuteResult := true
+		CLICK_HELPER(myStep.getTapX(), myStep.getTapY())
+	}
+
+	if(myExecuteResult){
+		mySleep := myStep.getMillisecondsWaitAfter()
+		Sleep, %mySleep%
+	}
+
+	return myExecuteResult
+}
+
+DISPLAY_MESSAGE(theMessage){
+	TrayTip ; hide previous
+	Sleep, 1
+	TrayTip, OCR_Action_Sequencer, %theMessage%, 1
+	;MsgBox, %theMessage%
+}
+
+GET_PREVIOUS_STEP_ORDINAL(theCurrentStepOrdinal, theStepTotal){
+	; 1-Indexed
+	myReturn := theCurrentStepOrdinal - 1
+	if(myReturn < 1){
+		myReturn := theStepTotal
+	}
+	return myReturn
+}
+
+EXECUTE_SEQUENCE_UNTIL_CAP(theSequenceDataStringRelativePath){
+	; This is the main driver operation
+
+	mySequenceDataString := READ_FILE_CONTENTS(theSequenceDataStringRelativePath)
+	mySequenceData := LOAD_SEQUENCE_DATA_FROM_JSON_STRING(mySequenceDataString)
+	myCap := mySequenceData.getSequenceLoopLimit()
+	myStepListSize := mySequenceData.getStepList().Length()
+
+	mySequenceLoopCounter := 0
+
+	while (mySequenceLoopCounter < myCap) {
+		mySequenceLoopCounter++
+		;DISPLAY_MESSAGE(mySequenceLoopCounter)
+
+		myCurrentStep := 0

```

Comment:
Let's add "Counter" to the end for clarity. Especially now that we have a proper "Step" object.